src/frontend/src/handler/create_table.rs

```diff
@@ -773,42 +928,70 @@ pub async fn handle_create_table(
         let context = OptimizerContext::from_handler_args(handler_args);
         let source_schema = check_create_table_with_source(context.with_options(), source_schema)?;
         let col_id_gen = ColumnIdGenerator::new_initial();
+        let properties = context.with_options().inner().clone().into_iter().collect();
 
-        let (plan, source, table) = match source_schema {
-            Some(source_schema) => {
+        let ((plan, source, table), job_type) = match (source_schema, cdc_source.as_ref()) {
+            (Some(source_schema), None) => (
                 gen_create_table_plan_with_source(
                     context,
                     table_name.clone(),
-                    columns,
+                    column_defs,
                     constraints,
                     source_schema,
                     source_watermarks,
                     col_id_gen,
                     append_only,
                 )
-                .await?
-            }
-            None => gen_create_table_plan(
-                context,
-                table_name.clone(),
-                columns,
-                constraints,
-                col_id_gen,
-                source_watermarks,
-                append_only,
-            )?,
-        };
+                .await?,
+                TableJobType::Normal,
+            ),
+            (None, None) => (
+                gen_create_table_plan(
+                    context,
+                    table_name.clone(),
+                    column_defs,
+                    constraints,
+                    col_id_gen,
+                    source_watermarks,
+                    append_only,
+                )?,
+                TableJobType::Normal,
+            ),
 
+            (None, Some(source_name)) => (
+                gen_create_table_plan_for_cdc_source(
+                    context.into(),
+                    source_name.clone(),
+                    table_name.clone(),
+                    external_table.unwrap_or(table_name.real_value()),
+                    column_defs,
+                    constraints,
+                    col_id_gen,
+                )?,
+                TableJobType::SharedCdcSource,
+            ),
+            (Some(_), Some(_)) => return Err(ErrorCode::NotSupported(
+                "Data format and encoding format doesn't apply to table created from a CDC source"
+                    .into(),
+                "Remove the FORMAT and ENCODE specification".into(),
+            )
+            .into()),
+        };
         let mut graph = build_graph(plan);
-        graph.parallelism = session
-            .config()
-            .get_streaming_parallelism()
-            .map(|parallelism| Parallelism { parallelism });
-        (graph, source, table)
+        graph.parallelism =
+            if cdc_source.is_some() || CdcTableType::from_properties(&properties).can_backfill() {
+                Some(Parallelism { parallelism: 1 })
```

Comment:
The parallelism of the table job is fixed to 1, because the `CdcBackfillExecutor` is single-threaded implementation. Would it cause any performance issue for downstream jobs derived from the table? cc @BugenZhao